plugins/sigma.parsers.cypher/sigma.parsers.cypher.js

```diff
@@ -0,0 +1,237 @@
+                    callback = sig;
+                    sig = null;
+                }
+
+                // Call the callback if specified:
+                if (callback)
+                    callback(sig || graph);
+            }
+        };
+
+        var postData = JSON.stringify(data);
+        xhr.setRequestHeader('Content-type', 'application/json; charset=utf-8');
+        if( login && password) {
+            xhr.setRequestHeader('Authorization', 'Basic ' + btoa(login + ':' + password));
+        }
+        xhr.send(postData);
+    };
+
+    /**
+     * This function parse a neo4j cypher query result.
+     *
+     * @param  {object}       neo4jResult The URL of neo4j server.
+     *
+     * @return A graph object
+     */
+    sigma.neo4j.cypher_parse = function(neo4jResult) {
+        var graph = { nodes: [], edges: [] },
+            nodesMap = {},
+            edgesMap = {},
+            key;
+
+        // Iteration on all result data
+        neo4jResult.results[0].data.forEach(function (data, index, ar) {
+
+            // iteration on graph for all node
+            data.graph.nodes.forEach(function (node, index, ar) {
+
+                var sigmaNode =  {
+                    id : node.id,
+                    label : node.id,
+                    x : Math.random(),
+                    y : Math.random(),
+                    size : 1,
+                    color : '#000000',
+                    neo4j_labels : node.labels,
+                    neo4j_data : node.properties
+                };
+
+                if (sigmaNode.id in nodesMap) {
+                    // do nothing
+                } else {
+                    nodesMap[sigmaNode.id] = sigmaNode;
+                }
+            });
+
+            // iteration on graph for all node
+            data.graph.relationships.forEach(function (edge, index, ar) {
+                var sigmaEdge =  {
+                    id : edge.id,
+                    label : edge.type,
+                    source : edge.startNode,
+                    target : edge.endNode,
+                    color : '#7D7C8E',
+                    neo4j_type : edge.type,
+                    neo4j_data : edge.properties
+                };
+
+                if (sigmaEdge.id in edgesMap) {
+                    // do nothing
+                } else {
+                    edgesMap[sigmaEdge.id] = sigmaEdge;
+                }
+            });
+
+        });
+
+        // construct sigma nodes
+        for (key in nodesMap) {
+            graph.nodes.push(nodesMap[key]);
+        }
+        // construct sigma nodes
+        for (key in edgesMap) {
+            graph.edges.push(edgesMap[key]);
+        }
+
+        return graph;
+    };
+
+    /**
+     * This function call neo4j to get all labels of the graph.
+     *
+     * @param  {string}       server      The URL of neo4j server.
+     *
+     * @return An array of label
+     */
+    sigma.neo4j.getLabels = function(server, callback) {
```

Comment:
No, getTypes & getLabels are not used in the plugin.But I think those methods can be usefull for users. 

What do you think ? We keep them or not ?